examples/lldpd.lua

```diff
@@ -0,0 +1,94 @@
+--
+-- SPDX-FileCopyrightText: (c) 2025 Ashwani Kumar Kamal <ashwanikamal.im421@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+-- Daemon to send LLDP frames on given interface
+local socket = require("socket")
+local linux  = require("linux")
+local thread = require("thread")
+
+local shouldstop = thread.shouldstop
+
+local ETH_P_ALL  = 0x0003
+local ETH_P_LLDP = 0x88cc
+
+local IFACE = "veth0"
+local ifindex = linux.ifindex(IFACE)
+
+-- LLDP multicast destination
+local LLDP_DST = string.char(0x01,0x80,0xc2,0x00,0x00,0x0e)
+
+local function get_src_mac()
+    local rx = socket.new(socket.af.PACKET, socket.sock.RAW, ETH_P_ALL)
+	rx:bind(string.pack(">H", ETH_P_ALL), ifindex)
+
+    local frame = rx:receive(2048)
+	rx:close()
+
+    return frame:sub(7, 12)
+end
+
+local function tlv(t, value)
+    return string.pack(">H", (t << 9) | #value) .. value
+end
+
+local function build_lldp_frame(src)
+    local chassis_tlv =
+        string.pack(">H", (1 << 9) | 6) .. src
+
+    local port_tlv = tlv(2, IFACE)
+
+    local ttl_tlv =
+        string.pack(">H", (3 << 9) | 2) ..
+        string.pack(">H", 120)
+
+	local sys_name_tlv = tlv(6, "lunatik-lldpd")
+
+	local sys_desc_tlv = tlv(6, "LLDP example implemented using lunatik")
+
+	-- system capabilities
+	-- https://datatracker.ietf.org/doc/html/rfc4957#section-3.4.3
+	local sys_caps = 0x0080  -- station only
+	local sys_caps_tlv =
+    string.pack(">H", (7 << 9) | 4) ..
+    string.pack(">H", sys_caps) ..
+    string.pack(">H", sys_caps)
+
+    local end_tlv = string.pack(">H", 0)
+
+	local lldp_payload =
+        chassis_tlv ..
+        port_tlv ..
+        ttl_tlv ..
+        sys_name_tlv ..
+        sys_desc_tlv ..
+		sys_caps_tlv ..
+        end_tlv
+
+    local eth_hdr =
+        LLDP_DST ..
+        src ..
+        string.pack(">H", ETH_P_LLDP)
+
+    return eth_hdr .. lldp_payload
+end
+
+local src_mac = get_src_mac()
+local lldp_frame = build_lldp_frame(src_mac)
+
+local function worker()
+    local tx <close> = socket.new(socket.af.PACKET, socket.sock.RAW, ETH_P_LLDP)
+    tx:bind(ifindex)
```

Comment:
this is a common pattern, perhaps it would be good to have a socket/raw.lua for that.. I've done something similar for unix domain sockets.. feel free to create a new issue for this if you want to tackle